crates/oci-distribution/src/client.rs

```diff
@@ -89,6 +89,19 @@ impl Client {
 
         let (manifest, digest) = self.pull_manifest(image).await?;
 
+        if manifest.layers.is_empty() {
+            return Err(anyhow::anyhow!("no layers to pull"));
+        }
+
+        for layer in &manifest.layers {
+            if layer.media_type != WASM_LAYER_MEDIA_TYPE {
```

Comment:
I am vigorously opposed to doing this here.  This crate should be a (moderately) general purpose OCI client - even the kubelet goes to lengths not to be bound to WASM - that is all at the provider level.

I get though that you don't want to go to the trouble of fetching the layers and then have the consumer find that they're the wrong type.  One option would be for the consumer to say what layer types it accepts.  Another would be to break up the flow so that the consumer gets to look at the list of layers and decide whether it wants to proceed, though that may be overkill unless there are other criteria or there is something a consumer can usefully do with individual layers rather than the whole image.

Reply:
I completely agree. oras follows a similar approach of the consumer informing oras which media types the client should accept, and validate against those accepted media types. I'm sure we can amend the API to accomodate that workflow. While it may introduce breaking changes to the API, it would allow us to validate incoming content without placing a huge restriction on what content `oci-distribution` can fetch.

Reply:
Okay. I managed to pull it out of the function call. The caller now has to pass a `Vec<&str>` of accepted media types to be pulled.